src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -99,4 +147,207 @@ private void saveSalt(CmpdRegMolecule mol) throws IOException, CmpdRegMolFormatE
 		}
 	}
 
+	@Transactional
+	public String exportSalts() {
+		CmpdRegSDFWriter sdfWriter = null;
+		try {
+			sdfWriter = sdfWriterFactory.getCmpdRegSDFBufferWriter();
+		} catch (IllegalArgumentException e1) {
+			e1.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		// Get List of Salts 
+		List<Salt> allSaltList = Salt.findAllSalts();
+
+		for (Salt salt : allSaltList) {
+			try {
+				CmpdRegMolecule currSalt = cmpdRegMoleculeFactory.getCmpdRegMolecule(salt.getMolStructure());
+				currSalt.setProperty("name", salt.getName());
+				currSalt.setProperty("abbrev", salt.getAbbrev());
+				currSalt.setProperty("formula", salt.getFormula());
+				currSalt.setProperty("molWeight", String.valueOf(salt.getMolWeight()));
+				currSalt.setProperty("charge", String.valueOf(salt.getCharge()));
+				sdfWriter.writeMol(currSalt);
+			} catch (CmpdRegMolFormatException e) {
+				logger.error("bad structure error: " + salt.getMolStructure());
+			} catch (IOException e) {
+				logger.error("IO error reading in molfile");
+				e.printStackTrace();
+			}
+
+		}
+		try {
+			sdfWriter.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		logger.debug("ou	tput SDF: " + sdfWriter.getBufferString());
+		return sdfWriter.getBufferString();
+	}
+
+	// Helper Method Used to Check Dependencies Within ACAS
+	private Map<String, HashSet<String>> checkACASDependencies(
```

Comment:
Brian Bolt did some work recently to refactor and simplify dependency checking. You should have a look at his code and chat with him, and see if there are places where you can reuse functions he's written, or follow that same pattern.

Reply:
This code was based off of code Brian B. messaged me a few weeks ago but I will follow up and see if it is still representative of what I should be doing.

Reply:
Implemented